### PR TITLE
Music: enable Amplitude analytics on standalone projects

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
## Warning!!

The [AP CSP Create Performance Task](https://apcentral.collegeboard.org/courses/ap-computer-science-principles/exam) is in progress. The most critical dates are from April 3 - April 30, 2024. Please consider any risk introduced by this PR that could affect our students taking AP CSP. Code.org students taking AP CSP primarily use App Lab for their Create Task, however a small percent use Game Lab. Carefully consider whether your change has any risk of alterering, changing, or breaking anything in these two labs. Even small changes, such as a different button color, are considered significant during this time period. Reach out to the Student Learning team or Curriculum team for more details.

## Description

Enable Amplitude analytics on standalone music projects (rather than /projectbeats)

Screenshot 2024-04-30 at 12 06 14 PM

## Links

https://codedotorg.atlassian.net/browse/LABS-727

## Testing story

Tested with standalone projects and script levels - confirmed that events are only emitted for standalone projects.

### CODE REVIEW

File: apps/src/music/analytics/AnalyticsReporter.ts [hunk 1]
```diff
@@ -171,29 +165,6 @@ export default class AnalyticsReporter {
...
-
-  onInstructionsVisited(page: number) {
-    if (!this.sessionInProgress) {
-      this.log('No session in progress');
-      return;
-    }
-
-    this.currentInstructionsPage = page;
-    this.maxInstructionsSeen = Math.max(this.maxInstructionsSeen, page);
-  }
```
Comment (sanchitmalhotra126) [inline apps/src/music/analytics/AnalyticsReporter.ts:195]: Removing some data we don't log any more

File: apps/src/music/analytics/AnalyticsReporter.ts [hunk 2]
```diff
@@ -267,6 +236,19 @@ export default class AnalyticsReporter {
   }
 
   log(message: string) {
-    console.log(`[AMPLITUDE ANALYTICS EVENT]: ${message}`);
+    console.log(`[MUSIC AMPLITUDE ANALYTICS EVENT]: ${message}`);
+  }
+
+  private formatUserId(userId: number) {
```
Comment (sanchitmalhotra126) [inline apps/src/music/analytics/AnalyticsReporter.ts:242]: Copied from https://github.com/code-dot-org/code-dot-org/blob/f280b95523fb6a3aabbe624250146256a9480cef/apps/src/lib/util/AnalyticsReporter.js#L126 I'd imagine later we might want to consolidate/replace with this metrics reporter, so just copying over for now

File: apps/src/music/analytics/AnalyticsReporter.ts [hunk 3]
```diff
@@ -267,6 +236,19 @@ export default class AnalyticsReporter {
...
+    const userIdString = userId.toString() || 'none';
+    if (!userId) {
+      return userIdString;
+    }
```
Comment (thomasoniii) [inline apps/src/music/analytics/AnalyticsReporter.ts]: non-blocking, but this feels backwards to me. Would it make more sense as:

```
if (!userId) {
  return 'none'
}
const userIdString = userId.toString()
```

My thinking is that we're explicitly assuming the `userId` exists (and is truthy) on line 243, but then on 244 we're checking to see if it doesn't exist (or, is at least non-truthy) and then bombing out later.

**Review by thomasoniii — APPROVED** [review]
another minor suggestion/opinion question, but otherwise lgtm 🚀

File: apps/src/music/analytics/AnalyticsReporter.ts (hunk 3 above)
Comment (sanchitmalhotra126) [reply apps/src/music/analytics/AnalyticsReporter.ts]: Oh yep good call - copied this from another class but didn't catch this, thanks

**Review by breville — APPROVED** [review]

**Review by breville — APPROVED** [review]

**Pull request merged**